### PR TITLE
Remap items in board

### PR DESCRIPTION
**Pull request by charlesBochet**
## Context

We are refactoring the record board which has been hard coded in the codebase to display opportunities. 

## What

In this PR:
- adding state select field type on opportunity standard object
- Introducing states to store columns: columnIds, columnFamilyState, columnFamilySelector (to ease usage), isColumnFirstFamilyState, isColumnLastFamilyState. We are introducing these familyState to guarantee good performance. A column should be able to read its columnInformation but do not re-render when other columns are changed. 
- using this state to compute board columns based on select field options (the first we find for now)
- Re-adding columnHeader, columnMenu components but only redirecting to Settings to edit Select options

image

## Next steps:
- Display them through basic cards
- re-apply viewFields
- pagination (performance)
- action bar, context menu, selection

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx [hunk 1]
```diff
@@ -2,38 +2,36 @@ import { useCallback, useEffect } from 'react';
...
   });
 
+  useLoadRecordIndexBoard(objectNameSingular, recordBoardId);
+
   const navigate = useNavigate();
 
   const navigateToSelectSettings = useCallback(() => {
-    navigate(`/settings/objects/${objectNamePlural}`);
-  }, [navigate, objectNamePlural]);
+    navigate(`/settings/objects/${objectNameSingular}`);
```
Comment (Weiko) [inline packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx:28]: Why this change? I think the URL is using plural form as of today, right?

**Review by Weiko — APPROVED** [review]
Left a comment otherwise lgtm

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx (hunk 1 above)
Comment (charlesBochet) [reply packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx:28]: You are right!
